### ISSUE TITLE
[Questions regarding Reproducing the CIFAR-10 Results (Discrete)]

### ISSUE THREAD

**Issue by chen-hao-chao** [facebookresearch/flow_matching#31]
Hi,

Thank you for your excellent work. I have been working on reproducing the results of the discrete flow-matching model on CIFAR-10 and made some adjustments to the training script. Specifically, I configured the script to use 4 GPUs and set the batch_size to 64 to maintain the same effective batch size of 256. Below are the submission commands I used:

```
python submitit_train.py \
--ngpus=4 \
--dataset=cifar10 \
--nodes=1 \
--discrete_flow_matching \
--batch_size=64 \
--accum_iter=1 \
--cfg_scale=0.0 \
--use_ema \
--epochs=3000 \
--class_drop_prob=1.0 \
--compute_fid \
--sym_func
```

I added `--ngpus=4 ` and modified `--batch_size=64`. With this setup, I achieved a final FID of 6.695 at the 2500th epoch, which differs from the results reported in the paper and README. I was wondering if you could provide any guidance on reproducing the experimental results accurately (based on a 4-GPU setup).

I also noticed that the experimental setup described in the paper uses an effective batch size of 512 and completes training over approximately 3072 epochs (calculated as 300K / (50000/512)=3072). Could this difference in setup be a factor?

I would greatly appreciate any advice or insights you can share. Thank you very much for your kind support!

**Comment by mhavasi** [comment]
Hey, thanks for the question.

The batch size (8x32=256) definitely reproduces the results reported here, so a 4 GPU setup with batch size 64 should also work.

When training for 3000 epochs, we compute the FID every 100 epochs and we found that the model reached the best FID=3.58 at epoch 2500. The FID at epoch 3000 was slightly higher than that at 2500.

Given that its different hardware, it good to ensure that you use float32 precision if possible, and double check that the gradients are correctly shared between the processes.

Otherwise, I don't see any issues with your setup.

**Comment by chen-hao-chao** [comment]
I reran the experiment using an 8-GPU setup and obtained an FID of 3.82 at the 2500th epoch. The observed loss values were of similar magnitudes (~2.8 at the 2500th epoch) compared to the 4-GPU setup. While I am still uncertain about the cause of performance degradation in the 4-GPU setup, the original configuration works.
